### PR TITLE
signumone-ks: init at 3.1.2

### PR DESCRIPTION
**Pull request by WolfangAukang**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/signumone-ks/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  libPath = stdenv.lib.makeLibraryPath buildInputs;
+
+  unpackPhase = ''
+    dpkg-deb -x ${src} ./
+  '';
+
+  installPhase = ''
+    DESKTOP_PATH=$out/share/applications/signumone-ks.desktop
+
+    mkdir -p $out/{bin,share/applications}
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/signumone-ks/default.nix:36]:
```suggestion
    mkdir -p $out/bin $out/share/applications
```

File: pkgs/applications/misc/signumone-ks/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    mv opt/SignumOne-KS/SignumOne-KS.desktop $DESKTOP_PATH
+    mv opt $out
+
+    sed -e "s|\(Exec=\)/opt/SignumOne-KS|\1$out/bin|g" -i $DESKTOP_PATH
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/signumone-ks/default.nix:40]: Please use substituteInPlace.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or torch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108191` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>signumone-ks</li>
</ul>
</details>

File: pkgs/applications/misc/signumone-ks/default.nix (hunk 2 above)
Comment (WolfangAukang) [reply pkgs/applications/misc/signumone-ks/default.nix]: Done in force-pushed commit

File: pkgs/applications/misc/signumone-ks/default.nix (hunk 1 above)
Comment (WolfangAukang) [reply pkgs/applications/misc/signumone-ks/default.nix]: Done

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108191` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>signumone-ks</li>
</ul>
</details>

File: pkgs/applications/misc/signumone-ks/default.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+    makeWrapper
+  ];
+
+  buildInputs = [
+    atk glibc gdk-pixbuf stdenv.cc.cc ffmpeg
+    libav_0_8 gtk3 libXtst
```
Comment (dotlambda) [inline pkgs/applications/misc/signumone-ks/default.nix:24]: I doubt we need both `ffmpeg` and `libav_0_8`.
Also, shouldn't `glibc` and `stdenv.cc.cc` automatically be added by `stdenv.mkDerivation`?

Comment (WolfangAukang) [reply pkgs/applications/misc/signumone-ks/default.nix:24]: Not adding libav_9_8 will not build.

```
autoPatchelfHook could not satisfy dependency libavcodec-ffmpeg.so.56 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-56.so
autoPatchelfHook could not satisfy dependency libavcodec.so.53 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-53.so
autoPatchelfHook could not satisfy dependency libavcodec.so.57 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-57.so
autoPatchelfHook could not satisfy dependency libavcodec.so.56 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-56.so
autoPatchelfHook could not satisfy dependency libavcodec.so.55 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-55.so
autoPatchelfHook could not satisfy dependency libavcodec.so.54 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-54.so
autoPatchelfHook could not satisfy dependency libavformat-ffmpeg.so.56 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-56.so
autoPatchelfHook could not satisfy dependency libavformat.so.54 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-54.so
autoPatchelfHook could not satisfy dependency libavformat.so.55 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-55.so
autoPatchelfHook could not satisfy dependency libavformat.so.56 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-56.so
autoPatchelfHook could not satisfy dependency libavformat.so.57 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-57.so
autoPatchelfHook could not satisfy dependency libavformat.so.53 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-53.so
/nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3
```

Comment (WolfangAukang) [reply pkgs/applications/misc/signumone-ks/default.nix:24]: Never mind, I forgot that it is ignoring missing deps. It is running as expected. Going to fix it.

Comment (dotlambda) [reply pkgs/applications/misc/signumone-ks/default.nix:24]: > Not adding `libav_0_8` will not build.
> 
> ```
> autoPatchelfHook could not satisfy dependency libavcodec-ffmpeg.so.56 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-56.so
> autoPatchelfHook could not satisfy dependency libavcodec.so.53 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-53.so
> autoPatchelfHook could not satisfy dependency libavcodec.so.57 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-57.so
> autoPatchelfHook could not satisfy dependency libavcodec.so.56 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-56.so
> autoPatchelfHook could not satisfy dependency libavcodec.so.55 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-55.so
> autoPatchelfHook could not satisfy dependency libavcodec.so.54 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-54.so
> autoPatchelfHook could not satisfy dependency libavformat-ffmpeg.so.56 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-56.so
> autoPatchelfHook could not satisfy dependency libavformat.so.54 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-54.so
> autoPatchelfHook could not satisfy dependency libavformat.so.55 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-55.so
> autoPatchelfHook could not satisfy dependency libavformat.so.56 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-56.so
> autoPatchelfHook could not satisfy dependency libavformat.so.57 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-57.so
> autoPatchelfHook could not satisfy dependency libavformat.so.53 wanted by /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-53.so
> /nix/store/zzzb60wwkcyq71n1h8gk6idjx66djzd6-signumone-ks-3.1.3
> ```

This probably means it requires `libav_0_8`, but not `ffmpeg`.

Comment (dotlambda) [reply pkgs/applications/misc/signumone-ks/default.nix:24]: Actually, you should be fine with ffmpeg_3:
```
$ nix-build -A ffmpeg_3
$ ls result/lib
libavcodec.so             libavfilter.so.6          libavresample.so.3.7.0   libswresample.so
libavcodec.so.57          libavfilter.so.6.107.100  libavutil.so             libswresample.so.2
libavcodec.so.57.107.100  libavformat.so            libavutil.so.55          libswresample.so.2.9.100
libavdevice.so            libavformat.so.57         libavutil.so.55.78.100   libswscale.so
libavdevice.so.57         libavformat.so.57.83.100  libpostproc.so           libswscale.so.4
libavdevice.so.57.10.100  libavresample.so          libpostproc.so.54        libswscale.so.4.8.100
libavfilter.so            libavresample.so.3        libpostproc.so.54.7.100
```

**Pull request merged**